### DISCUSSION TITLE
How to loop video with gstreammer for rtsp output ?

### DISCUSSION THREAD

**Discussion by thanhlnbka** [bluenviron/mediamtx#1124] [Questions]
I use the command " gst-launch-1.0 rtspclientsink name=s location=rtsp://localhost:8554/mystream filesrc location=file.mp4 ! qtdemux name=d d.video_0 ! queue ! s.sink_0 d.audio_0 ! queue ! s.sink_1" to output rtsp, but without looping the video.
I need your help, thanks for reading !

**Comment by aler9** [comment]
Hello, GStreamer doesn't support looping mp4 files, you have to first convert the file to a container format that supports concatenation (like MPEG-TS or MPEG-2) and the use `multifilesrc`, please read here:

https://stackoverflow.com/questions/6833147/looping-a-video-with-gstreamer-and-gst-launch

[a comment by an automated account is not shown]
